### PR TITLE
Several small improvements in the network module

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

File: cnd/src/http_api/problem.rs [hunk 1]
```diff
@@ -54,11 +53,6 @@ pub fn from_anyhow(e: anyhow::Error) -> HttpApiProblem {
         return HttpApiProblem::new("Swap not found.").set_status(StatusCode::NOT_FOUND);
     }
 
-    if e.downcast_ref::<SwapExists>().is_some() {
```
Comment (D4nte) [inline cnd/src/http_api/problem.rs:57]: Are we not sending an error specific to existing swap anymore?
Can we please re-introduce the functionality?

File: cnd/src/network/comit/swaps.rs [hunk 2]
```diff
@@ -21,10 +21,12 @@ pub enum Error {
...
 
-/// Tracks the state of a swap in the communication phase, T is
-/// ReplySubstream<NegotiatedSubstream>
+/// Tracks the state of a swap in the communication phase.
+///
+/// We use a type parameter for the substream type so we can write unit tests
+/// without creating actual substreams.
 #[derive(Debug)]
-pub struct Swaps<T> {
+pub struct Swaps<T = ReplySubstream<NegotiatedSubstream>> {
```
Comment (D4nte) [inline cnd/src/network/comit/swaps.rs:29]: Nice trick.

File: cnd/src/http_api/problem.rs (hunk 1 above)
Comment (thomaseizinger) [reply cnd/src/http_api/problem.rs:57]: The error struct was never used outside of this line so I deleted it. Meaning there is no change in functionality with this PR :)

I am not sure this PR is the best place to introduce this functionality again.

I am gonna re-open https://github.com/comit-network/comit-rs/issues/2490.

**Pull request merged**